app/Foliage/PreparePackageVersion.hs

```diff
@@ -60,23 +65,28 @@ preparePackageVersion inputDir metaFile = do
     readPackageVersionSpec' (inputDir </> metaFile) >>= \case
       PackageVersionSpec {packageVersionRevisions, packageVersionTimestamp = Nothing}
         | not (null packageVersionRevisions) -> do
-            error $
-              unlines
-                [ inputDir </> metaFile <> " has cabal file revisions but the original package has no timestamp.",
-                  "This combination doesn't make sense. Either add a timestamp on the original package or remove the revisions"
-                ]
+          error $
+            unlines
+              [ inputDir </> metaFile <> " has cabal file revisions but the original package has no timestamp.",
+                "This combination doesn't make sense. Either add a timestamp on the original package or remove the revisions"
+              ]
       PackageVersionSpec {packageVersionRevisions, packageVersionTimestamp = Just pkgTs}
         | any ((< pkgTs) . revisionTimestamp) packageVersionRevisions -> do
-            error $
-              unlines
-                [ inputDir </> metaFile <> " has a revision with timestamp earlier than the package itself.",
-                  "Adjust the timestamps so that all revisions come after the original package"
-                ]
+          error $
+            unlines
+              [ inputDir </> metaFile <> " has a revision with timestamp earlier than the package itself.",
+                "Adjust the timestamps so that all revisions come after the original package"
+              ]
       meta ->
         return meta
 
   srcDir <- prepareSource pkgId pkgSpec
 
+  -- FIXME: This produce a Shake error since it `need` the file:
```

Comment:
Okay, so what we've learned is that this actually depends on the "final" cabal file, which depends on doing the revision computation. It's unclear why we do things in this order, but potentially you need to move the revision sequence calculation here. You could then put the computed actual revision files in `cabalFileRevisions`. 

I'm not 100% sure what the right thing is here, because the split between `preparePackageVersion` and the later code is unclear to me (@andreabedini ? maybe we could get some haddock? :) ). But I _think_ `preparePackageVersion` is supposed to pre-compute stuff we need later, so having it build all the intermediary revised cabal files seems reasonable.

Reply:
Yes, that sounds very sensible indeed.